### PR TITLE
Gen migrations

### PR DESCRIPTION
**Pull request by jakedt**
### Description of Changes

* details about the implementation of the changes
* motivation for the change (broken code, new feature, etc)
* contrast with previous behavior

#### Changes:

* ..
* ..

#### Issue: <link to story or task>

**TESTING** ->

**BREAKING CHANGE** ->

---

## Reviewer Checklist

- [ ] It works!
- [ ] Comments provide sufficient explanations for the next contributor
- [ ] Tests cover changes and corner cases
- [ ] Follows Quay syntax patterns and format

### CODE REVIEW

File: data/migrations/dba_operator.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import yaml
```
Comment (josephschorr) [inline data/migrations/dba_operator.py]: Can you add a doc comment linking to the DBA operator and explaining what this module does in relation?

File: data/migrations/dba_operator.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    def dump_yaml_and_reset(self, stream, revision, down_revision):
+        migration_object = {
+            "apiVersion": "dbaoperator.app-sre.redhat.com/v1alpha1",
+            "kind": "DatabaseMigration",
+            "metadata": {"name": revision,},
+            "spec": {
+                "previous": down_revision,
+                "migrationContainerSpec": {
+                    "name": revision,
+                    "image": "quay.io/quaymigrations/realtest",
```
Comment (josephschorr) [inline data/migrations/dba_operator.py]: What is `quay.io/quaymigrations/realtest`?

File: data/migrations/dba_operator.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                },
+                "schemaHints": self._schema_hints,
+            },
+        }
+        yaml.dump(migration_object, stream)
+        self._schema_hints = []
+
+    def add_hints_from_ops(self, directive):
+        for op in directive.ops:
+            if isinstance(op, ops.ModifyTableOps):
```
Comment (josephschorr) [inline data/migrations/dba_operator.py]: I wonder if we could make this easier to read via some sort of table of lambdas... not sure

File: data/migrations/dba_operator.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    def hint_drop_index(self, index_name, table_name, **kwargs):
+        self._schema_hints.append(
+            {"operation": "dropIndex", "table": table_name, "indexName": index_name,}
+        )
+
+
+class OpLogger(object):
+    def __init__(self, delegate_module, migration):
+        self._delegate_module = delegate_module
+        # self._migration = migration
```
Comment (josephschorr) [inline data/migrations/dba_operator.py]: Not needed anymore?

File: data/migrations/env.py [hunk 5]
```diff
@@ -164,6 +185,17 @@ def run_migrations_online():
     finally:
         connection.close()
 
+    for (revision, previous_revision), migration in revision_to_migration.items():
+        write_migration(migration, revision, previous_revision)
+
+
+def write_migration(migration, revision, previous_revision):
```
Comment (josephschorr) [inline data/migrations/env.py]: Can we name this slightly differently, given that it writes the DBA operator migration stuff only?

File: data/migrations/migration.sh [hunk 6]
```diff
@@ -75,7 +75,7 @@ down_postgres() {
 
 gen_migrate() {
   # Generate a database with the schema as defined by the existing alembic model.
-  QUAY_OVERRIDE_CONFIG=$1 PYTHONPATH=. alembic upgrade head
+  QUAY_OVERRIDE_CONFIG=$1 PYTHONPATH=. alembic -x writemigrations=true upgrade head
```
Comment (josephschorr) [inline data/migrations/migration.sh]: ditto. Maybe... `-generatedbaopmigrations=true`?

File: data/migrations/dba_operator.py (hunk 2 above)
Comment (jakedt) [reply data/migrations/dba_operator.py]: I guess I should make that config, and set it to `quay.io/quay/quay` for now? I wonder if it should be config to the migration generator or to the dba operator.

File: data/migrations/dba_operator.py (hunk 3 above)
Comment (jakedt) [reply data/migrations/dba_operator.py]: `if` statements are the worst, amirite?

File: data/migrations/dba_operator.py (hunk 4 above)
Comment (jakedt) [reply data/migrations/dba_operator.py]: Nope. Removed.

File: data/migrations/env.py (hunk 5 above)
Comment (jakedt) [reply data/migrations/env.py]: Done

File: data/migrations/migration.sh (hunk 6 above)
Comment (jakedt) [reply data/migrations/migration.sh]: Done.

File: data/migrations/dba_operator.py (hunk 1 above)
Comment (jakedt) [reply data/migrations/dba_operator.py]: Added

**Comment by jakedt** [comment]
@josephschorr ping

**Pull request merged**